diff --git a/src/shaders/skybox_shader.rs b/src/shaders/skybox_shader.rs
index 5726fc3..015885b 100644
--- a/src/shaders/skybox_shader.rs
+++ b/src/shaders/skybox_shader.rs
@@ -29,11 +29,13 @@ pub struct SkyboxShader {
 
 impl SkyboxShader {
     pub fn new(vertex_file: &str, fragment_file: &str) -> Self {
-        Self {
+        let mut result = Self {
             base: ShaderBase::new(vertex_file, fragment_file),
             location_projection: 0,
             location_view: 0
-        }
+        };
+        result.get_uniforms();
+        result
     }
 
     pub fn load_view_matrix(&self, view_matrix: &glm::TMat4<f32>) {
@@ -51,11 +53,13 @@ impl SkyboxShader {
 
 impl Default for SkyboxShader {
     fn default() -> Self {
-        Self {
+        let mut result = Self {
             base: ShaderBase::new("Skybox", "Skybox"),
             location_projection: 0,
             location_view: 0
-        }
+        };
+        result.get_uniforms();
+        result
     }
 }
 
@@ -68,4 +72,4 @@ impl Shader for SkyboxShader {
             self.location_view = gl::GetUniformLocation(self.base.id, c_string.as_ptr());
         }
     }
-}
\ No newline at end of file
+}
