diff --git a/src/renderer/render_master.rs b/src/renderer/render_master.rs
index 3920ee3..80d2196 100644
--- a/src/renderer/render_master.rs
+++ b/src/renderer/render_master.rs
@@ -75,11 +75,15 @@ impl RenderMaster {
         if self.draw_box {
             unsafe {
                 gl::Disable(gl::CULL_FACE);
+                gl::DepthFunc(gl::LEQUAL);
             }
             self.skybox_renderer.render(camera);
+            unsafe {
+                gl::DepthFunc(gl::LESS);
+            }
             self.draw_box = false;
         }
 
         window.display();
     }
-}
\ No newline at end of file
+}
